### ISSUE TITLE
Moving feeds/items between channels

### ISSUE THREAD

**Issue by fluffy-critter** [indieweb/microsub#40]
Is there a mechanism for moving a feed and/or its items between channels, ideally preserving their items' read/unread states?

What about having a feed's items going into different channels based on various rules (e.g. `<category>` tags in Atom)?

One possible use case is on e.g. my website, I have different categories which serve different purposes, and would benefit from different UX for each one, and it would be nice to have some sort of mechanism to not have to subscribe to each site section under a different channel.

Some of this functionality would be better-expressed by the clients/endpoints rather than Microsub itself but it's not clear where the line should be drawn or how to provide a good UX for it.

Perhaps this could be a case for having a post-update hook where an external filter can take action on an item as it comes in before it is presented to the reader.

**Comment by aaronpk** [comment]
I've had this on my todo list for Aperture for quite a while: https://github.com/aaronpk/Aperture/issues/37

Moving a feed/subscription to a new channel does seem like something potentially useful for a client to do, although I'm not sure what I would want the UX of that to look like.

Before this ends up in the Microsub spec, I'd want to see at least Aperture and one other server support the feature on its own even without a client UI for it.

If any client developers have an idea for what this might look like in the apps that'd be great to see sketched out here as well.

Thanks for starting this discussion!

**Comment by mblaney** [comment]
I have a UI to switch channels based on author, so the feed defaults to one channel and specific authors in that feed get moved to other channels (think facebook and twitter feeds). This is on unicyclic.com, which is currently only a microsub server, but it's on my list to make the client side of my reader a microsub client.

**Comment by swentel** [comment]
I have a case for being able to move single items from one channel to another (to my 'Read later' / 'Saved' channel). I'll do an experiment with it on the Drupal side of things and Indigenous for Android. Will ping back when it's in. 

API proposal

- action=timeline
- method=move
- channel={uid}
- entry={entry-id} or entry[]={entry-id}

The 'channel' is the channel where you want to move this post to on the server.

As for moving feeds from one channel to another, that's interesting too from a client perspective. Drupal Indieweb UI supports it (and moves items over to the other channel as well), so it I don't have to code much there anyway. Indigenous supports adding/removing/ordering channels and adding/removing feeds, just not moving them to another channel, so that would be a nice addition. I'll experiment with that as well.

API proposal

- action=follow
- method=update
- channel={uid}
- url=feed

Basically the same as follow, but with an additional method 'update'.

**Comment by swentel** [comment]
Moving post(s) from one channel to another is now supported on server level in the Drupal IndieWeb module (using the properties from proposal above) and in Indigenous for Android.

**Comment by swentel** [comment]
Moving a feed from one channel to another is now supported on server level in the Drupal IndieWeb module (using the properties from proposal above) and in Indigenous for Android.
